eng/common/TestResources/New-TestResources.ps1

```diff
@@ -15,6 +15,9 @@ param (
     [ValidatePattern('^[-a-zA-Z0-9\.\(\)_]{0,80}(?<=[a-zA-Z0-9\(\)])$')]
     [string] $BaseName,
 
+    [ValidatePattern('^[-\w\._\(\)]+$')]
+    [string] $ResourceGroupName,
```

Comment:
@heaths - We have a scenario here where we want to force resources to be deployed to the same resource group. The `New-TestResources.ps1` script is called multiple times on different service directories and all resources end up deployed in the same place. We think this will make debugging smoke tests and cleanup easier overall.

Reply:
Why is this being changed in a language-specific PR? @weshaggard asked that all changes to it be in the azure-sdk-tools repo.

Reply:
The changes I'm making to smoke tests depend on these changes so we're making them together in this repo. Once changes are finalized here I'll open the corresponding PR in the tools repo.